src/System.Memory/src/System/SpanExtensions.cs

```diff
@@ -274,5 +274,107 @@ public static void CopyTo<T>(this T[] array, Span<T> destination)
         {
             new ReadOnlySpan<T>(array).CopyTo(destination);
         }
+
+        /// <summary>
+        /// Determines whether two sequences overlap.
+        /// </summary>
+        public static unsafe bool Overlaps<T>(this Span<T> first, ReadOnlySpan<T> second)
+        {
+            if (second.IsEmpty)
+                return false;
+
+            ref T firstRef = ref first.DangerousGetPinnableReference();
+            ref T secondRef = ref second.DangerousGetPinnableReference();
+
+            IntPtr firstByteCount = Unsafe.ByteOffset(ref firstRef, ref Unsafe.Add(ref firstRef, first.Length));
+            IntPtr secondByteCount = Unsafe.ByteOffset(ref secondRef, ref Unsafe.Add(ref secondRef, second.Length));
+
+            IntPtr diff = Unsafe.ByteOffset(ref firstRef, ref secondRef);
+
+            return (sizeof(IntPtr) == sizeof(int))
+                ? (uint)diff < (uint)firstByteCount || (uint)diff > (uint)-(int)secondByteCount
+                : (ulong)diff < (ulong)firstByteCount || (ulong)diff > (ulong)-(long)secondByteCount;
+        }
+
+        /// <summary>
+        /// Determines whether two sequences overlap and outputs the element offset.
+        /// </summary>
+        public static unsafe bool Overlaps<T>(this Span<T> first, ReadOnlySpan<T> second, out int elementOffset)
+        {
+            if (second.IsEmpty)
```

Comment:
@ektrah according to the proposal checking for `IsEmpty` is wrong. Simply omit it.

Reply:
I still wonder how there can be an overlap with an empty span. But I guess you won...

Reply:
cc @terrajobst

Reply:
> I still wonder how there can be an overlap with an empty span.

It is similar to first being empty but still overlapping with second (see case H):
 - **H**: `first` is empty within `second`
```
first:    [)
          xRef
second:  [--------------------------)     
         yRef                       yRef + yLength
```


If second is empty and doesn't overlap with first, Overlaps should return false (similar to if first is empty - case I/J).